llmfoundry/data/packing.py

```diff
@@ -485,6 +485,8 @@ def profile_packing(
             tmp_path_to_broadcast = tempfile.TemporaryDirectory().name
             gathered_paths = dist.all_gather_object(tmp_path_to_broadcast)
             tmp_path = gathered_paths[local_rank_zero]
+            if stream_config.get('remote') is None:
```

Comment:
I might be missing something, but I don't think this is the right fix. Shouldn't the fix be to just not overwrite the local dir if it is provided and already exists? (and in the above if statement too)

Reply:
That works too. This change also works because remote gets copied into local.

I can't repro this bug with remote unset and local set + autopacking (the above if statement)

Reply:
oh thats odd that it only exists when using streams...well anyway, they should probably both do the same thing, and that should probably be to not overwrite local if its specified?

Reply:
Sure, my hesitation is because I don't know the original motivation for using tmp paths, but if you think this is a safe change then 👍